src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java

```diff
@@ -56,16 +56,12 @@ public boolean canResolve(@NotNull final IRequestManager manager, final IRequest
         if (!manager.getColony().getWorld().isRemote)
         {
             final Colony colony = (Colony) manager.getColony();
-            final Set<TileEntityWareHouse> wareHouses = getWareHousesInColony(colony);
+            final Set<BuildingWareHouse> wareHouses = getWareHousesInColony(colony);
             wareHouses.removeIf(Objects::isNull);
 
-            try
+            if (!wareHouses.isEmpty())
             {
-                return wareHouses.stream().anyMatch(wareHouse -> wareHouse.hasMatchingItemStackInWarehouse(itemStack -> requestToCheck.getRequest().matches(itemStack), requestToCheck.getRequest().getCount()));
-            }
-            catch (Exception e)
-            {
-                Log.getLogger().error(e);
+                return true;
```

Comment:
This change breaks the RS Logic, please revert. The canResolve phase is exactly there to check if a Warehouse has a potential stack, returning true here breaks the entire logic and would result in the RS never actually checking any provider that has a lower priority, even if no warehouse has the requested item.

BLOCKING CHANGE!

Reply:
hm from what I can see this is the only place where it is used: https://github.com/ldtteam/minecolonies/blob/e9413d09d2434eeaddbe23b52989e845102fb674/src/main/java/com/minecolonies/coremod/colony/requestsystem/management/handlers/RequestHandler.java#L150

So it should be fine to fail on the attempt instead of the canresolve check right?

Reply:
It needs to crash on the resolve() call if it returns false on the canResolve() call. It needs to succeed if canResolve() returned true.